pkg/image/sif.go

```diff
@@ -82,24 +107,26 @@ func (f *sifFormat) initializer(img *Image, fileinfo os.FileInfo) error {
 			continue
 		}
 
+		// checks if the partition length is greater that the file
+		// size which may reveal a corrupted image (see issue #3996)
+		if fimg.Filesize < desc.Filelen {
```

Comment:
Shouldn't this be `desc.Filelen + desc.Fileoff` or is that already included?

Reply:
Yes and thanks for the review and for spotting my stupid mistakes, now it's time for me to end this work day :)

Reply:
Sure :)
May I suggest adding tests that running `singularity inspect` or `singularity shell/exec` on an incomplete image returns a meaningful error message? I guess there are already tests for running those commands on correct images. To create an incomplete image I'd take a normal one and chop of the last byte. Maybe additionally tests where you destroy the squashfs header. just to make sure this code is correctly triggered and to avoid regressions.

Reply:
I will add missing unit tests for SIF images in a first time in this PR, and create another PR for e2e tests